### PR TITLE
C++: Add `_Exit` to the list of exiting (non-returning) functions

### PR DESCRIPTION
**Pull request by jketema**

### CODE REVIEW

File: cpp/ql/lib/DefaultOptions.qll [hunk 1]
```diff
@@ -62,7 +62,7 @@ class Options extends string {
     f.getASpecifier().hasName("noreturn")
     or
     f.hasGlobalOrStdName([
-        "exit", "_exit", "abort", "__assert_fail", "longjmp", "__builtin_unreachable"
+        "exit", "_exit", "_Exit", "abort", "__assert_fail", "longjmp", "__builtin_unreachable"
```
Comment (geoffw0) [inline cpp/ql/lib/DefaultOptions.qll:65]: I'm curious why "longjmp" is on this list.  It's not the same as the others.  I suppose it exits the current position in code, never to return to where it left off.

(I don't expect you to have an answer)

**Review by geoffw0 — APPROVED** [review]
LGTM

File: cpp/ql/lib/DefaultOptions.qll (hunk 1 above)
Comment (jketema) [reply cpp/ql/lib/DefaultOptions.qll:65]: Indeed, I don't have an answer.

Comment (jketema) [reply cpp/ql/lib/DefaultOptions.qll:65]: I agree that we should probably rename the predicate. I'm also a bit hesitant to do so given that it lives in  `DefaultOptions.qll`, so it's a bit difficult to gauge the impact of this outside of the CodeQL repo.

**Pull request merged**